REST-API-ADVANCED/advanced/src/main/java/com/epam/esm/user/entity/authoritydeserializer/CustomAuthorityDeserializer.java

```diff
@@ -0,0 +1,27 @@
+package com.epam.esm.user.entity.authoritydeserializer;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+
+import java.io.IOException;
+import java.util.LinkedList;
+import java.util.List;
+
+public class CustomAuthorityDeserializer extends JsonDeserializer<List<GrantedAuthority>> {
+    @Override
+    public List<GrantedAuthority> deserialize(JsonParser jp, DeserializationContext ctxt) throws IOException {
+        ObjectMapper mapper = (ObjectMapper) jp.getCodec();
+        JsonNode jsonNode = mapper.readTree(jp);
+        List<GrantedAuthority> grantedAuthorities = new LinkedList<>();
```

Comment:
Why LinkedList?

Reply:
I used it by mistake. In any case, it shouldn't have affected performance in any way cuz list contains one or two items